Userland/Utilities/init.cpp

```diff
@@ -127,26 +137,65 @@ static ErrorOr<void> reopen_base_file_descriptors()
     // This affects also every other process that inherits the file descriptors
     // from SystemServer, so it is important for other things (also for ProcFS
     // tests that are running in CI mode).
-    int stdin_new_fd = TRY(Core::System::open("/dev/null"sv, O_NONBLOCK));
+    int stdin_new_fd = TRY(acquire_new_stdin_fd(emergency));
     TRY(Core::System::dup2(stdin_new_fd, 0));
     TRY(Core::System::dup2(stdin_new_fd, 1));
     TRY(Core::System::dup2(stdin_new_fd, 2));
     return {};
 }
 
-ErrorOr<int> serenity_main(Main::Arguments)
+static ErrorOr<void> execute_emergency_shell()
+{
+    outln("Emergency mode: Dropping to emergency shell mode");
+    outln("You may use this shell as rescue environment now.");
+    if (!Core::System::access("/bin/Shell"sv, X_OK, 0).is_error())
+        TRY(Core::System::exec("/bin/Shell"sv, Vector { "/bin/Shell"sv }, Core::System::SearchInPath::No));
+    if (!Core::System::access("/bin/BuggieBox"sv, X_OK, 0).is_error())
+        TRY(Core::System::exec("/bin/BuggieBox"sv, Vector { "/bin/BuggieBox"sv, "emergency_shell"sv }, Core::System::SearchInPath::No));
```

Comment:
This will not work. I should probably rename `emergency_shell` to `/bin/Shell` or just `sh`?